Improve plugin docs to make clear that JsonToIon requires JSONL format (with new-line)

### Expected Behavior

JsonToIon task should work seamlessly.

### Actual Behaviour

JsonToIon task leads to an error.
Here is the error stacktrace:

```
2024-06-10 09:21:50.020 com.fasterxml.jackson.core.io.JsonEOFException: Unexpected end-of-input: expected close marker for Array (start marker at [Source: REDACTED (`StreamReadFeature.INCLUDE_SOURCE_IN_LOCATION` disabled); line: 1, column: 1])
 at [Source: REDACTED (`StreamReadFeature.INCLUDE_SOURCE_IN_LOCATION` disabled); line: 1, column: 2]
2024-06-10 09:21:50.020 Unexpected end-of-input: expected close marker for Array (start marker at [Source: REDACTED (`StreamReadFeature.INCLUDE_SOURCE_IN_LOCATION` disabled); line: 1, column: 1])
 at [Source: REDACTED (`StreamReadFeature.INCLUDE_SOURCE_IN_LOCATION` disabled); line: 1, column: 2]
2024-06-10 09:21:50.020 reactor.core.Exceptions$ReactiveException: com.fasterxml.jackson.core.io.JsonEOFException: Unexpected end-of-input: expected close marker for Array (start marker at [Source: REDACTED (`StreamReadFeature.INCLUDE_SOURCE_IN_LOCATION` disabled); line: 1, column: 1])
 at [Source: REDACTED (`StreamReadFeature.INCLUDE_SOURCE_IN_LOCATION` disabled); line: 1, column: 2]
	at reactor.core.Exceptions.propagate(Exceptions.java:410)
	at reactor.core.publisher.BlockingSingleSubscriber.blockingGet(BlockingSingleSubscriber.java:102)
	at reactor.core.publisher.Mono.block(Mono.java:1779)
	at io.kestra.plugin.serdes.json.JsonToIon.run(JsonToIon.java:85)
	at io.kestra.plugin.serdes.json.JsonToIon.run(JsonToIon.java:33)
	at io.kestra.core.runners.WorkerTaskThread.doRun(WorkerTaskThread.java:77)
	at io.kestra.core.runners.AbstractWorkerThread.run(AbstractWorkerThread.java:56)
	Suppressed: java.lang.Exception: #block terminated with an error
		at reactor.core.publisher.BlockingSingleSubscriber.blockingGet(BlockingSingleSubscriber.java:104)
		... 5 more
Caused by: com.fasterxml.jackson.core.io.JsonEOFException: Unexpected end-of-input: expected close marker for Array (start marker at [Source: REDACTED (`StreamReadFeature.INCLUDE_SOURCE_IN_LOCATION` disabled); line: 1, column: 1])
 at [Source: REDACTED (`StreamReadFeature.INCLUDE_SOURCE_IN_LOCATION` disabled); line: 1, column: 2]
	at com.fasterxml.jackson.core.base.ParserMinimalBase._reportInvalidEOF(ParserMinimalBase.java:699)
	at com.fasterxml.jackson.core.base.ParserBase._handleEOF(ParserBase.java:514)
	at com.fasterxml.jackson.core.base.ParserBase._eofAsNextChar(ParserBase.java:531)
	at com.fasterxml.jackson.core.json.ReaderBasedJsonParser._skipWSOrEnd(ReaderBasedJsonParser.java:2524)
	at com.fasterxml.jackson.core.json.ReaderBasedJsonParser.nextToken(ReaderBasedJsonParser.java:705)
	at com.fasterxml.jackson.databind.deser.std.UntypedObjectDeserializerNR._deserializeNR(UntypedObjectDeserializerNR.java:281)
	at com.fasterxml.jackson.databind.deser.std.UntypedObjectDeserializerNR.deserialize(UntypedObjectDeserializerNR.java:79)
	at com.fasterxml.jackson.databind.deser.DefaultDeserializationContext.readRootValue(DefaultDeserializationContext.java:342)
	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4899)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3846)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3814)
	at io.kestra.plugin.serdes.json.JsonToIon.lambda$nextRow$1(JsonToIon.java:118)
	at io.kestra.core.utils.Rethrow.lambda$throwConsumer$0(Rethrow.java:50)
	at reactor.core.publisher.FluxCreate.subscribe(FluxCreate.java:97)
	at reactor.core.publisher.Mono.subscribe(Mono.java:4568)
	at reactor.core.publisher.Mono.block(Mono.java:1778)
	... 4 more
```

**Findings:**
We are able to convert such JSON files to ION seamlessly:
```
{"product_id":"1","product_name":"streamline turn-key systems","product_category":"Electronics","brand":"gomez"}
{"product_id":"2","product_name":"morph viral applications","product_category":"Household","brand":"wolfe"}
{"product_id":"3","product_name":"expedite front-end schemas","product_category":"Household","brand":"davis-martinez"}
```

We are NOT able to convert a file containing list of records as JSON:
```
[
    {"product_id":"1","product_name":"streamline turn-key systems","product_category":"Electronics","brand":"gomez"},
    {"product_id":"2","product_name":"morph viral applications","product_category":"Household","brand":"wolfe"},
    {"product_id":"3","product_name":"expedite front-end schemas","product_category":"Household","brand":"davis-martinez"}
]
```

### Steps To Reproduce

1. Run the flow given the example flow.

### Environment Information

- Kestra Version: 0.17.0
- Plugin version: 0.17.0
- Operating System (OS / Docker / Kubernetes): Docker
- Java Version (If not docker):


### Example flow

```
id: json_to_ion
namespace: dev

tasks:
  - id: http_download
    type: io.kestra.plugin.core.http.Download
    uri: https://huggingface.co/datasets/kestra/datasets/raw/main/json/app_events.json

  - id: to_ion
    type: io.kestra.plugin.serdes.json.JsonToIon
    from: "{{ outputs.http_download.uri }}"
```

--- a/src/main/java/io/kestra/plugin/serdes/json/JsonToIon.java
+++ b/src/main/java/io/kestra/plugin/serdes/json/JsonToIon.java
@@ -38,18 +38,18 @@ import static io.kestra.core.utils.Rethrow.throwConsumer;
 @Getter
 @NoArgsConstructor
 @Schema(
-    title = "Read a json file and write it to an ion serialized data file.",
+    title = "Read a JSON file and write it to an ION serialized data file.",
     description = """
-    Please note that we support JSONL format only, i.e. one JSON per line.
+    Please note that we support JSONL format only, i.e. one JSON dictionary/map per line.
 
-    A sample JSON file content would be:
+    Here is how a sample JSON file content might look like:
     ```
     {"product_id":"1","product_name":"streamline turn-key systems","product_category":"Electronics","brand":"gomez"},
     {"product_id":"2","product_name":"morph viral applications","product_category":"Household","brand":"wolfe"},
     {"product_id":"3","product_name":"expedite front-end schemas","product_category":"Household","brand":"davis-martinez"}
     ```
 
-    We do NOT support an array of JSON objects, i.e. the JSON file with the following content is not supported:
+    We do NOT support an array of JSON objects. A JSON file in the following array format is not supported:
     ```
     [
         {"product_id":"1","product_name":"streamline turn-key systems","product_category":"Electronics","brand":"gomez"},

--- a/src/main/java/io/kestra/plugin/serdes/json/JsonToIon.java
+++ b/src/main/java/io/kestra/plugin/serdes/json/JsonToIon.java
@@ -38,7 +38,26 @@ import static io.kestra.core.utils.Rethrow.throwConsumer;
 @Getter
 @NoArgsConstructor
 @Schema(
-    title = "Read a json file and write it to an ion serialized data file."
+    title = "Read a json file and write it to an ion serialized data file.",
+    description = """
+    Please note that we support JSONL format only, i.e. one JSON per line.
+
+    A sample JSON file content would be:
+    ```
+    {"product_id":"1","product_name":"streamline turn-key systems","product_category":"Electronics","brand":"gomez"},
+    {"product_id":"2","product_name":"morph viral applications","product_category":"Household","brand":"wolfe"},
+    {"product_id":"3","product_name":"expedite front-end schemas","product_category":"Household","brand":"davis-martinez"}
+    ```
+
+    We do NOT support an array of JSON objects, i.e. the JSON file with the following content is not supported:
+    ```
+    [
+        {"product_id":"1","product_name":"streamline turn-key systems","product_category":"Electronics","brand":"gomez"},
+        {"product_id":"2","product_name":"morph viral applications","product_category":"Household","brand":"wolfe"},
+        {"product_id":"3","product_name":"expedite front-end schemas","product_category":"Household","brand":"davis-martinez"}
+    ]
+    ```
+    """
 )
 @Plugin(
     examples = {

